src/main/java/com/abitmorecode/songrest/Services/SongService.java

```diff
@@ -0,0 +1,220 @@
+package com.abitmorecode.songrest.Services;
+
+import com.abitmorecode.songrest.Controller.SongController;
+import com.abitmorecode.songrest.Models.Song;
+import com.abitmorecode.songrest.SongControllerException.NoIdAvailableException;
+import com.abitmorecode.songrest.SongControllerException.SameSongAlreadyExistException;
+import com.abitmorecode.songrest.SongControllerException.SongDoesntExistException;
+import com.abitmorecode.songrest.SongControllerException.SongIdAlreadyExistException;
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Song Controller (Service)
+ */
+
+@Service
```

Comment:
Bisschen nitpicky. https://stackoverflow.com/questions/1440096/difference-between-repository-and-service

Reply:
Sollte normalerweise eine Repository Class nicht eher eine Schnitstelle zu einer Art Datenbank sein?

Reply:
Naja nicht umbedingt Datenbank. Datenquelle. In eurem Fall habt ihr im Grunde eine In-Memory Datenbank implementiert